lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -380,6 +386,34 @@ class ReportUIFeatures {
     this._document.body.removeChild(a);
     setTimeout(_ => URL.revokeObjectURL(href), 500);
   }
+
+  _handleStickyHeader() {
+    const topbarEl = this._document.querySelector('.lh-topbar');
+    const scoreScaleEl = this._document.querySelector('.lh-scorescale');
+    const stickyHeaderEl = this._document.querySelector('.lh-sticky-header');
+    const highlightEl = this._document.querySelector('.lh-highlighter');
+    if (!topbarEl || !scoreScaleEl || !stickyHeaderEl || !highlightEl) return;
+
+    // Show sticky header when the score scale begins to go underneath the topbar.
+    const showStickyHeader =
+      topbarEl.getBoundingClientRect().bottom - scoreScaleEl.getBoundingClientRect().top >= 0;
+    stickyHeaderEl.classList.toggle('lh-sticky-header--visible', showStickyHeader);
+
+    // Highlight mini gauge when section is in view.
+    // In view = the last category that starts above the middle of the window.
+    const categoryEls = Array.from(this._document.querySelectorAll('.lh-category'));
+    const categoriesAboveTheMiddle =
+      categoryEls.filter(el => el.getBoundingClientRect().top - window.innerHeight / 2 < 0);
+    const highlightIndex =
+      categoriesAboveTheMiddle.length > 0 ? categoriesAboveTheMiddle.length - 1 : 0;
+
+    // Category order matches gauge order in sticky header.
+    // TODO(hoten): not 100% true yet, need to order gauges like: core, pwa, plugins. Remove
+    // this comment when that is done.
+    const gaugeToHighlight = stickyHeaderEl.querySelectorAll('.lh-gauge__wrapper')[highlightIndex];
+    (/** @type {HTMLElement} */ (highlightEl)).style.left =
```

Comment:
Ah yeah I guess `querySelector` can return things that aren't HTMLElements, bummer! Maybe make the assertion at the top where it's declared?

Reply:
just refactored to use _dom + make member vars